crates/rustc_codegen_spirv/src/spirv_type.rs

```diff
@@ -248,6 +249,20 @@ impl SpirvType {
             ),
             Self::Sampler => cx.emit_global().type_sampler(),
             Self::SampledImage { image_type } => cx.emit_global().type_sampled_image(image_type),
+
+            Self::InterfaceBlock { inner_type } => {
+                let mut emit = cx.emit_global();
+                let id = emit.id();
+                let result = emit.type_struct_id(Some(id), iter::once(inner_type));
+                emit.decorate(result, Decoration::Block, iter::empty());
+                emit.member_decorate(
+                    result,
+                    0,
+                    Decoration::Offset,
+                    [Operand::LiteralInt32(0)].iter().cloned(),
```

Comment:
What happens if there's multiple interface blocks in the function? Won't this point them to the same offset?

Reply:
This is just what you would get from this kind of struct:
```rust
#[spirv(block)]
struct InterfaceBlock<T> {
    inner: T,
}
```

So this isn't a regression, and I believe offsets are per interface variable.

There's an unrelated issue that we allow multiple push constants and multiple other bindings with the same `(descriptor_set, binding)`, that's the only situations I can think of in which anything would overlap.